src/asynk/async_worker.rs

```diff
@@ -106,4 +100,223 @@ where
             };
         }
     }
+    #[cfg(test)]
+    pub async fn run_tasks_until_none(&mut self) -> Result<(), Error> {
+        loop {
+            match self
+                .queue
+                .fetch_and_touch_task(&self.task_type.clone())
+                .await
+            {
+                Ok(Some(task)) => {
+                    self.sleep_params.maybe_reset_sleep_period();
+                    self.run(task).await?
+                }
+                Ok(None) => {
+                    return Ok(());
+                }
+                Err(error) => {
+                    error!("Failed to fetch a task {:?}", error);
+
+                    self.sleep().await;
+                }
+            };
+        }
+    }
+}
+
+#[cfg(test)]
+mod async_worker_tests {
+    use super::AsyncWorker;
+    use crate::asynk::async_queue::AsyncQueueTest;
+    use crate::asynk::async_queue::AsyncQueueable;
+    use crate::asynk::async_queue::FangTaskState;
+    use crate::asynk::AsyncRunnable;
+    use crate::asynk::Error;
+    use crate::RetentionMode;
+    //use crate::SleepParams;
```

Comment:
let's remove unused imports

Reply:
done in [1945344](https://github.com/ayrat555/fang/pull/46/commits/194534459c67e042eff605b1b1798a260ca277f8)